lib/ansible/plugins/inventory/vmware_inventory.py

```diff
@@ -0,0 +1,337 @@
+# Copyright: (c) 2018, Ansible Project
+# Copyright: (c) 2018, Abhijeet Kasurde <akasurde@redhat.com>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+DOCUMENTATION = '''
+    name: vmware
+    plugin_type: inventory
+    short_description: VMware inventory source
+    description:
+        - Get inventory hosts from VMware environment.
+        - Uses any file which ends with vmware.yml or vmware.yaml as a YAML configuration file.
+    extends_documentation_fragment:
+      - constructed
+      - inventory_cache
+    options:
+        hostname:
+            description: Name of vCenter or ESXi server
+            env:
+              - name: VMWARE_SERVER
+        username:
+            description: Name of vSphere admin user
+            env:
+              - name: VMWARE_USERNAME
+        password:
+            description: Password of vSphere admin user
+            env:
+              - name: VMWARE_PASSWORD
+        port:
+            description: Port number used to connect to vCenter or ESXi Server
+            env:
+              - name: VMWARE_PORT
+        validate_certs:
+            description:
+            - Allows connection when SSL certificates are not valid. Set to C(false) when certificates are not trusted.
+            default: True
+            type: boolean
+        with_tags:
+            description:
+            - Include tags and associated virtual machines.
+            - Requires 'vSphere Automation SDK' and 'vCloud Suite SDK' libraries to be installed on the given controller machine.
+            - Please refer following URLs for installation steps
+            - 'https://code.vmware.com/web/sdk/65/vsphere-automation-python'
+            - 'https://code.vmware.com/web/sdk/60/vcloudsuite-python'
+            default: False
+            type: boolean
+'''
+
+EXAMPLES = '''
+simple_config_file:
+
+    plugin: vmware
+    strict: False
+    hostname: 10.65.223.31
+    username: administrator@vsphere.local
+    password: Esxi@123$%
+    validate_certs: False
+    with_tags: True
+
+'''
+
+import ssl
+import atexit
+
+try:
+    # requests is required for exception handling of the ConnectionError
+    import requests
+    HAS_REQUESTS = True
```

Comment:
check for specific version, requests behaviour and features are very version dependent

Reply:
Done.